### DISCUSSION TITLE
Add support for `css` prop on custom elements

### DISCUSSION THREAD

**Discussion by overbalance** [emotion-js/emotion#2883] [Q&A] [answered]
**The problem**

Using the `css` prop on custom elements yields an html element with a `css` attribute instead of `class`.

Input 
```html
<my-component css={css`color:red`}></my-component>
```

Current output 
```html
<style data-emotion="css" data-s="">.css-q3dxqh{color:red;}</style>
<my-component css="css-q3dxqh"></my-component>
```

**Proposed solution**

Detect `-` in elements and output `class="css-123456"`

Expected output 
```html
<style data-emotion="css" data-s="">.css-q3dxqh{color:red;}</style>
<my-component class="css-q3dxqh"></my-component>
```

**Additional context**

The only issue I have observed for this exact case is the incorrect attribute. There was discussion in 2018 that may be relevant. https://github.com/emotion-js/emotion/issues/1071 

I'd love to help but not sure exactly where to start.

**Comment by srmagura** [comment]
Which package are you importing the `css` function from?

**Comment by overbalance** [comment]
@srmagura I'm using ```import { css } from "@emotion/css";```

**Comment by srmagura** [comment]
When using the `css` prop, you must import the `css` function from `@emotion/react`. 

Also, it looks as though you have not configured the `@emotion/react` JSX runtime. The exact way to do this depends on the compiler you are using. Are you using Babel, tsc, swc, or something else?

**Reply by overbalance** [reply]
ESbuild via Remix btw

**Comment by Andarist** [comment, marked as answer]
Note that if you are using web components and you are only interested in using the `@emotion/css` package then you can just use the `class` prop:
```html
<my-component class={css`color:red`}></my-component>
```
